File: pkg/types/page.go
package types

type List struct {
	Items     []ListItem `json:"items,omitempty"`
	Dynamic   bool       `json:"dynamic,omitempty"`
	EmptyText string     `json:"emptyText,omitempty"`
	Actions   []Action   `json:"actions,omitempty"`
}

type ListItem struct {
	Id          string   `json:"id,omitempty"`
	Title       string   `json:"title"`
	Subtitle    string   `json:"subtitle,omitempty"`
	Accessories []string `json:"accessories,omitempty"`
	Actions     []Action `json:"actions,omitempty"`
}

type Detail struct {
	Actions []Action `json:"actions,omitempty"`
	Format  Format   `json:"format,omitempty"`
	Text    string   `json:"text,omitempty"`
}

type Format string

const (
	MarkdownFormat Format = "markdown"
	ANSIFormat     Format = "ansi"
	TemplateFormat Format = "template"
)

File: typescript/pkg/page.ts
import type { Action } from "./action.ts";

export type List = {
  items?: ListItem[];
  actions?: Action[];
  showDetail?: boolean;
  autoRefreshSeconds?: number;
  emptyText?: string;
};

export type Detail = {
  text?: string;
  markdown?: string;
  actions?: Action[];
};

export type ListItem = {
  title: string;
  id?: string;
  subtitle?: string;
  accessories?: string[];
  detail?: { text: string; } | { markdown: string; }
  actions?: Action[];
};
